superset/db_engine_specs/bigquery.py

```diff
@@ -72,11 +72,13 @@ class BigQueryEngineSpec(BaseEngineSpec):
     }
 
     @classmethod
-    def convert_dttm(cls, target_type: str, dttm: datetime) -> str:
+    def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
         tt = target_type.upper()
         if tt == "DATE":
-            return "'{}'".format(dttm.strftime("%Y-%m-%d"))
-        return "'{}'".format(dttm.strftime("%Y-%m-%d %H:%M:%S"))
+            return f"CAST('{dttm.isoformat()[:10]}' AS DATE)"
+        if tt == "TIMESTAMP":
```

Comment:
@betodealmeida do you know why the `TIMESTAMP` type wasn't defined for BigQuery?